File: BmiCalc/app/src/main/java/pamo/bmicalc/ui/quiz/QuizViewModel.java
package pamo.bmicalc.ui.quiz;

import android.util.Log;

import androidx.lifecycle.ViewModel;

import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;
import java.io.InputStream;

public class QuizViewModel extends ViewModel {

    public QuizViewModel() {
        Log.v("QuizViewModel", "Entered -> QuizViewModel constructor.");
    }
    public QuizQuestionsModel runParser(InputStream inputStream) throws XmlPullParserException, IOException{
        Log.v("QuizViewModel", "Entered -> runParser");
        QuestionsXMLparser qxp = new QuestionsXMLparser(inputStream);
        qxp.parseXML();

        return qxp.getQqm();
    }
}

File: BmiCalcKotlin/app/src/main/java/pamo/bmicalc_kotlin/ui/quiz/QuizViewModel.kt
package pamo.bmicalc_kotlin.ui.quiz

import android.util.Log
import androidx.lifecycle.ViewModel
import org.xmlpull.v1.XmlPullParserException
import java.io.IOException
import java.io.InputStream

class QuizViewModel : ViewModel() {
    private var qxp: QuestionsXMLparser? = null
    @Throws(XmlPullParserException::class, IOException::class)
    fun runParser(inputStream: InputStream?): QuizQuestionsModel? {
        Log.v("QuizViewModel", "Entered -> runParser")
        qxp = QuestionsXMLparser(inputStream!!)
        qxp!!.parseXML()
        return qxp!!.getQqm()
    }

    init {
        Log.v("QuizViewModel", "Entered -> QuizViewModel constructor.")
    }
}